src/execution_tree/primitives/column_set.cpp

```diff
@@ -94,25 +94,30 @@ namespace phylanx { namespace execution_tree { namespace primitives
                 {
                     actual_stop = stop;
                 }
-                else //(stop < 0)
+                else    //(stop < 0)
                 {
                     actual_stop = array_length + stop;
                 }
 
-                std::vector<int> result;
+                std::vector<std::int64_t> result;
 
                 if (step > 0)
                 {
-                    result.reserve((actual_stop - actual_start + step)/step);
-                    for (int i = actual_start; i < actual_stop; i += step)
+                    HPX_ASSERT(actual_stop > actual_start);
```

Comment:
Good move! However, is it ensured that this pre-condition is met? Do you check anywhere (and throw an exception), if not?

Reply:
@hkaiser I was hoping the assert would be enough to let the user know that something is wrong with his parameter selection. One thing I could do is convert the assert into an exception. What would you suggest ?

Reply:
The difference between an assert and an exception is that assert's are used to make sure pre-conditions are _met_. Exceptions are used to _report_ problems (for instance pre-condition violations). As a guideline, use exceptions in user-facing code, and assertions in code that is used internally only (i.e. not directly user-facing, assuming that the checks have been performed elsewhere). In your case I'd use an assertion in the function (as you have it) but add an explicit parameter check on some level above it.